requirements/connectors/weaviate.in

```diff
@@ -1,3 +1,5 @@
 -c ../common/constraints.txt
 
-weaviate-client
+# NOTE: Python client v3 weaviate.Client connections and methods are deprecated and will be removed by 2024-11-30
+# TODO: Upgrade to v4
```

Comment:
Does this mean we'll need to update this again in a couple weeks? If so, why not do the update now?

Reply:
Yep, so according to their documentation (https://weaviate.io/developers/weaviate/client-libraries/python/v3_v4_migration):

> The v4 Weaviate Python client API is a complete rewrite, aimed at an improved overall user experience. It is therefore also very different to the v3 API, and will require re-learning of changed patterns in the way you interact with Weaviate.

And here https://weaviate.io/developers/weaviate/client-libraries/python/python_v3:

> Starting in December 2024, the v4 [client installations](https://pypi.org/project/weaviate-client/) will no longer include the v3 API (i.e. the weaviate.Client class). This will help us to provide the best developer experience for you, support for the latest features, and clearly separate the two.

> The v3 client will continue to get critical security updates and bugfixes for the foreseeable future, but it will not support any new features.

So it looked like a lot of work, that is not needed (the code will continue to function, provided we downgrade client), and we have higher priorities (bug bash, adding more connectors) than doing this kind of "code quality" improvements :) But it probably makes sense to at least make a ticket for that..